### PR TITLE
feat: implement `InputSearch` component

### PR DESCRIPTION
**Pull request by HellWolf93**
TODO:
- [ ] Add Enter key icon

### CODE REVIEW

**Comment by LeandroTorresSicilia** [comment]
when press ESC button the input should clear its value

File: packages/search/src/components/InputSearch/__test__/inputSearch.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { Application } from 'react-rainbow-components';
+import InputSearch from '..';
+
+describe('<Trailing />', () => {
```
Comment (LeandroTorresSicilia) [inline packages/search/src/components/InputSearch/__test__/inputSearch.spec.js:6]: <InputSearch />

File: packages/search/src/components/InputSearch/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    const trailingRef = useRef<HTMLDivElement>(null);
+    const inputRef = useRef<InputComponent>(null);
+    const [paddingRight, setPaddingRight] = useState<number>();
+
+    useLayoutEffect(() => {
+        if (trailingRef.current) {
+            const { width } = trailingRef.current.getBoundingClientRect();
+            if (paddingRight !== width) setPaddingRight(width);
+        }
+    }, [paddingRight, value]);
```
Comment (LeandroTorresSicilia) [inline packages/search/src/components/InputSearch/index.tsx:27]: I think we can compute in a functional way the padding needed based on the props (value and variant) and then avoid using a state for the padding

File: packages/search/src/components/InputSearch/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            <StyledInput
+                name={name}
+                icon={<MagnifyingGlass />}
+                paddingRight={paddingRight}
+                value={value}
+                disabled={disabled}
+                readOnly={readOnly}
+                onChange={handleChange}
+                onKeyDown={handleKeyDown}
+                ref={inputRef}
```
Comment (TahimiLeonBravo) [inline packages/search/src/components/InputSearch/index.tsx:53]: to have all the functionality of a native search input we should pass it `type ="search"`

**Review by TahimiLeonBravo — CHANGES_REQUESTED** [review]

**Pull request merged**